UnitySDK/Assets/ML-Agents/Scripts/Grpc/GrpcExtensions.cs

```diff
@@ -0,0 +1,115 @@
+                Done = ai.done,
+                Id = ai.id,
+                CustomObservation = ai.customObservation
+            };
+            if (ai.memories != null)
+            {
+                agentInfoProto.Memories.Add(ai.memories);
+            }
+
+            if (ai.actionMasks != null)
+            {
+                agentInfoProto.ActionMask.AddRange(ai.actionMasks);
+            }
+
+            foreach (var obs in ai.visualObservations)
+            {
+                agentInfoProto.VisualObservations.Add(
+                    ByteString.CopyFrom(obs.EncodeToPNG())
+                );
+            }
+            return agentInfoProto;
+        }
+        
+        /// <summary>
+        /// Converts a Brain into to a Protobuff BrainInfoProto so it can be sent
+        /// </summary>
+        /// <returns>The BrainInfoProto generated.</returns>
+        /// <param name="name">The name of the brain.</param>
+        /// <param name="isTraining">Whether or not the Brain is training.</param>
+        public static BrainParametersProto ToProto(this BrainParameters bp, string name, bool isTraining)
+        {
+            var brainParametersProto = new BrainParametersProto
+            {
+                VectorObservationSize = bp.vectorObservationSize,
+                NumStackedVectorObservations = bp.numStackedVectorObservations,
+                VectorActionSize = {bp.vectorActionSize},
+                VectorActionSpaceType =
+                    (SpaceTypeProto)bp.vectorActionSpaceType,
+                BrainName = name,
+                IsTraining = isTraining
+            };
+            brainParametersProto.VectorActionDescriptions.AddRange(bp.vectorActionDescriptions);
+            foreach (var res in bp.cameraResolutions)
+            {
+                brainParametersProto.CameraResolutions.Add(
+                    new ResolutionProto
+                    {
+                        Width = res.width,
+                        Height = res.height,
+                        GrayScale = res.blackAndWhite
+                    });
+            }
+
+            return brainParametersProto;
+        }
+
+        /// <summary>
+        /// Convert metadata object to proto object.
+        /// </summary>
+        public static DemonstrationMetaProto ToProto(this DemonstrationMetaData dm)
+        {
+            var demoProto = new DemonstrationMetaProto
+            {
+                ApiVersion = DemonstrationMetaData.ApiVersion,
+                MeanReward = dm.meanReward,
+                NumberSteps = dm.numberExperiences,
+                NumberEpisodes = dm.numberEpisodes,
+                DemonstrationName = dm.demonstrationName
+            };
+            return demoProto;
+        }
+        
+        /// <summary>
+        /// Initialize metadata values based on proto object.
+        /// </summary>
+        public static DemonstrationMetaData FromProto(this DemonstrationMetaProto demoProto)
```

Comment:
Maybe `ToDemonstrationMetaData`? `ToManaged`? not sure what else to call it though.

Reply:
sounds good.